crates/emitter/src/ast_emitter.rs

```diff
@@ -127,8 +142,29 @@ impl<'alloc, 'opt> AstEmitter<'alloc, 'opt> {
             Statement::ForOfStatement { .. } => {
                 return Err(EmitError::NotImplemented("TODO: ForOfStatement"));
             }
-            Statement::ForStatement { .. } => {
-                return Err(EmitError::NotImplemented("TODO: ForStatement"));
+            Statement::ForStatement {
+                init,
+                test,
+                update,
+                block,
+                ..
+            } => {
+                CForEmitter {
+                    init,
+                    test,
+                    update,
```

Comment:
how about naming AST node field `maybe_init` etc, and naming the closure field `init` instead?

(the existing helpers are not much consistent tho...   we can rename them later)

the possible rules: 
  * if the field is non-optional single value, just the field name  
    `FooEmitter { key: |emitter| { ... } }`
  * if the field is optional single value, `maybe_*` for value and the field name for closure  
    `FooEmitter { maybe_key: field, key: |emitter, key| { ... } }`
  * if the filed is an array, plural for the array and singular for closure  
    `FooEmitter { keys: field, key: |emitter, key| { ... } }`

let me know if you have any opinion.

Reply:
this looks like a good solution to me and follows the BlockEmitter.